### PR TITLE
fix(hil-serl): discrete critic send through network

### PR DESCRIPTION
**Pull request by AdilZouitine**
## What this does
Explain what this PR does. Feel free to tag your PR with the appropriate label(s).

Examples:
|  Title               | Label           |
|----------------------|-----------------|
| Fixes #[issue]       | (🐛 Bug)        |
| Adds new dataset     | (🗃️ Dataset)    |
| Optimizes something  | (⚡️ Performance) |

## How it was tested
Explain/show how you tested your changes.

Examples:
- Added `test_something` in `tests/test_stuff.py`.
- Added `new_feature` and checked that training converges with policy X on dataset/environment Y.
- Optimized `some_function`, it now runs X times faster than previously.

## How to checkout & try? (for the reviewer)
Provide a simple way for the reviewer to try out your changes.

Examples:
```bash
pytest -sx tests/test_stuff.py::test_something
```
```bash
python -m lerobot.scripts.train --some.option=true
```

## SECTION TO REMOVE BEFORE SUBMITTING YOUR PR
**Note**: Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR. Try to avoid tagging more than 3 people.

**Note**: Before submitting this PR, please read the [contributor guideline](https://github.com/huggingface/lerobot/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr).

### CODE REVIEW

File: src/lerobot/scripts/rl/actor.py [hunk 1]
```diff
@@ -642,9 +642,19 @@ def update_policy_parameters(policy: SACPolicy, parameters_queue: Queue, device)
...
-        state_dict = move_state_dict_to_device(state_dict, device=device)
-        policy.load_state_dict(state_dict)
+        state_dicts = bytes_to_state_dict(bytes_state_dict)
+
+        # Load actor state dict
+        actor_state_dict = move_state_dict_to_device(state_dicts["policy"], device=device)
+        policy.actor.load_state_dict(actor_state_dict)
+
+        # Load discrete critic if present
+        if "discrete_critic" in state_dicts:
```
[the remark this answers is not in the document]
Comment (AdilZouitine) [reply src/lerobot/scripts/rl/actor.py]: fixed in afacf6763426f869678d757d131673022fc2a03b

**Comment by s1lent4gnt** [comment]
This line needs to be changed https://github.com/huggingface/lerobot/blob/user/azouitine/2025-7-8-fix-discrete-critic/src/lerobot/scripts/rl/actor.py#L320 to this 

`update_policy_parameters(policy=policy, parameters_queue=parameters_queue, device=device)`

cc @AdilZouitine

**Comment by AdilZouitine** [comment]
> This line needs to be changed https://github.com/huggingface/lerobot/blob/user/azouitine/2025-7-8-fix-discrete-critic/src/lerobot/scripts/rl/actor.py#L320 to this
> 
> `update_policy_parameters(policy=policy, parameters_queue=parameters_queue, device=device)`
> 
> cc @AdilZouitine

very good catch done in [c8c4e4c](https://github.com/huggingface/lerobot/pull/1468/commits/c8c4e4c4f2b032cbbb5677bb9d350790d33aa687)

**Comment by michel-aractingi** [comment]
LGTM

**Review by michel-aractingi — APPROVED** [review]
